src/raftstore/store/worker/region.rs

```diff
@@ -250,22 +364,93 @@ impl SnapContext {
         timer.observe_duration();
     }
 
-    fn handle_destroy(&mut self, region_id: u64, start_key: Vec<u8>, end_key: Vec<u8>) {
-        info!(
-            "[region {}] deleting data in [{}, {})",
-            region_id,
-            escape(&start_key),
-            escape(&end_key)
-        );
+    fn cleanup_range(
+        &mut self,
+        region_id: u64,
+        start_key: Vec<u8>,
+        end_key: Vec<u8>,
+        use_delete_files: bool,
+    ) {
+        if use_delete_files {
+            if let Err(e) = util::delete_all_files_in_range(&self.kv_db, &start_key, &end_key) {
+                error!(
+                    "[region {}] failed to delete files in [{}, {}): {:?}",
+                    region_id,
+                    escape(&start_key),
+                    escape(&end_key),
+                    e
+                );
+                return;
+            }
+        }
         if let Err(e) =
             util::delete_all_in_range(&self.kv_db, &start_key, &end_key, self.use_delete_range)
         {
             error!(
-                "failed to delete data in [{}, {}): {:?}",
+                "[region {}] failed to delete data in [{}, {}): {:?}",
+                region_id,
                 escape(&start_key),
                 escape(&end_key),
                 e
             );
+        } else {
+            info!(
+                "[region {}] succeed in deleting data in [{}, {})",
+                region_id,
+                escape(&start_key),
+                escape(&end_key),
+            );
+        }
+    }
+
+    // if clean_stale_peer_delay is 0, it means we don't use DeleteFilesInRange to clean range,
+    // so there are no overlap ranges, return false; otherwise return true
```

Comment:
Do you think the meaning of the return type is proper?

Reply:
Why it's not proper? `false` means we didn't clean overlap ranges.